File: Stack.cpp
#include "header.h"
#include "node.h"

// Реализация метода добавления элемента в стек
void Stack::push(std::string value) {
    Node* node = new Node{ value, top }; // Устанавливаем новый узел поверх стека
    top = node;
}

// Реализация метода удаления элемента из стека
void Stack::pop() {
    if (top == nullptr) {
        cout << "Стек пуст!" << endl;
    }
    else {
        Node* temp = top;
        top = top->next; // Убираем верхний элемент
        cout << "Удаленный элемент: " << temp->data << endl;
        delete temp;
    }
}

// Реализация метода вывода элементов стека
void Stack::display() {
    if (top == nullptr) {
        cout << "Стек пуст!" << endl;
    }
    else {
        Node* current = top;
        while (current != nullptr) {
            cout << current->data << " ";
            current = current->next;
        }
        cout << endl;
    }
}

File: go_lr1/stack.go
package main

import "fmt"

type Stack struct {
	top *Node1
}

func (s *Stack) Push(value string) {
	node := &Node1{data: value, next: s.top}
	s.top = node
}

func (s *Stack) Pop() {
	if s.top == nil {
		fmt.Println("Стек пуст!")
	} else {
		temp := s.top
		s.top = s.top.next
		fmt.Println("Удаленный элемент:", temp.data)
	}
}

func (s *Stack) Display() {
	if s.top == nil {
		fmt.Println("Стек пуст!")
	} else {
		current := s.top
		for current != nil {
			fmt.Print(current.data, " ")
			current = current.next
		}
		fmt.Println()
	}
}
